nixos/modules/config/locale.nix

```diff
@@ -37,12 +39,44 @@ in
       };
 
     };
+
+    location = {
+
+      latitude = mkOption {
+        type = types.nullOr types.float;
+        default = null;
+        description = ''
+          Your current latitude, between
+          <literal>-90.0</literal> and <literal>90.0</literal>. Must be provided
+          along with longitude.
+        '';
+      };
+
+      longitude = mkOption {
+        type = types.nullOr types.float;
+        default = null;
+        description = ''
+          Your current longitude, between
+          between <literal>-180.0</literal> and <literal>180.0</literal>. Must be
+          provided along with latitude.
+        '';
+      };
+
+    };
   };
 
   config = {
 
+    assertions = [
+      { assertion = lcfg.latitude != null && lcfg.longitude != null ||
+                    lcfg.latitude == null && lcfg.longitude == null;
+        message = "Either specify location.latitude and location.longitude or none at all."; }
+    ];
+
     environment.sessionVariables.TZDIR = "/etc/zoneinfo";
 
+    services.geoclue2.enable = mkDefault (lcfg.latitude == null && lcfg.longitude == null);
```

Comment:
You need to still have an option `provider`, defaulting to `"manual"`, and use that here and in the modules. How it is now, geoclue would be enabled for almost every NixOS user.

Reply:
So it ends up being something like `provider == "geoclue2" && latitude == null && longitude == null`?

Reply:
I think for this here just `services.geoclue2.enable = mkIf (provider == "geoclue") true`

(not `services.geoclue2.enable = provider == "geoclue"`, because not having the provider set to "geoclue" shouldn't disallow having geoclue enabled)

(and not with a `mkDefault` because using geoclue as a provider has to enable geoclue, otherwise it wouldn't work, it's not just some default, it's needed to work)